docs/dev-setup/JAVAChaincode.md

```diff
@@ -0,0 +1,71 @@
+# Java chaincode 
+
+Note: This guide generally assumes you have followed the development environment setup tutorial [here](https://github.com/hyperledger/fabric/blob/master/docs/API/SandboxSetup.md).
+## To get started developing Java chaincode: 
+1. Ensure you have gradle
+  * Download the binary distribution from [http://gradle.org/gradle-download/](http://gradle.org/gradle-download/)
+  * Unpack, move to the desired location, and add gradle's bin directory to your system path
+  * Ensure `gradle -v` works from the command-line, and shows version 2.12 or greater
+  * Optionally, enable the [gradle daemon](https://docs.gradle.org/current/userguide/gradle_daemon.html) for faster builds
+2. Ensure you have the Java 1.8 **JDK** installed. Also ensure Java's directory is on your path with `java -version`
+  * Additionally, you will need to have the [`JAVA HOME`](https://docs.oracle.com/cd/E19182-01/821-0917/6nluh6gq9/index.html) variable set to your **JDK** installation in your system path
+3. From your command line terminal, move to the `devenv` subdirectory of your workspace environment. Log into a Vagrant terminal by executing the following command:
+
+    vagrant ssh
+
+4. Build and run the peer process. To enable security and privacy after setting `security.enabled` and `security.privacy` settings to `true`.
+
+    cd $GOPATH/src/github.com/hyperledger/fabric
+    make peer
+    peer node start
+
+5. Change to Java shim root folder,
+
+	cd $GOPATH/src/github.com/hyperledger/fabric/core/chaincode/shim/java
+
+ and run 'gradle build'
```

Comment:
@xspeedcruiser step 5) is not necessary for running the java chaincode in docker (ie, not in "--peer-chaincodedev" mode). 

In fact, we need to add instructions to run in "--peer-chaincodedev" mode. To do that we will have to install gradle and java 1.8 in vagrant.